packages/@aws-cdk/aws-cloudfront/lib/edge-function.ts

```diff
@@ -0,0 +1,209 @@
+import * as cloudwatch from '@aws-cdk/aws-cloudwatch';
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as iam from '@aws-cdk/aws-iam';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as ssm from '@aws-cdk/aws-ssm';
+import {
+  App, BootstraplessSynthesizer, Construct, ConstructNode, DefaultStackSynthesizer, IStackSynthesizer,
+  ResourceEnvironment, Stack, StackProps, Token,
+} from '@aws-cdk/core';
+import * as resources from '@aws-cdk/custom-resources';
+
+/** Properties for creating a Lambda@Edge function */
+export interface EdgeFunctionProps extends lambda.FunctionProps { }
+
+/**
+ * A Lambda@Edge function.
+ *
+ * Convenience construct for requesting a Lambda function in the 'us-east-1' region for use with Lambda@Edge.
+ * Implements several restrictions enforced by Lambda@Edge.
+ */
+export class EdgeFunction extends Construct implements lambda.IVersion {
```

Comment:
Resource instead of Construct, maybe? Should remove some of the boiler plate around `stack` and `env`.

```suggestion
export class EdgeFunction extends Resource implements lambda.IVersion {
```

Reply:
Yup. I debated back and forth, and I think originally interpreted the Design Guidelines guidelines in an overly strict manner. I'll switch back to `Resource`.